src/cluster_legacy.c

```diff
@@ -6095,11 +6095,17 @@ sds clusterGenNodeDescription(client *c, clusterNode *node, int tls_primary) {
      * we are migrating to other instances or importing from other
      * instances. */
     if (node->flags & CLUSTER_NODE_MYSELF) {
+        char nodename[CLUSTER_NAMELEN + 1];
+
         for (j = 0; j < CLUSTER_SLOTS; j++) {
             if (server.cluster->migrating_slots_to[j]) {
-                ci = sdscatprintf(ci, " [%d->-%.40s]", j, server.cluster->migrating_slots_to[j]->name);
+                memcpy(nodename, server.cluster->migrating_slots_to[j]->name, CLUSTER_NAMELEN);
+                nodename[CLUSTER_NAMELEN] = '\0';
+                ci = sdscatfmt(ci, " [%i->-%s]", j, nodename);
```

Comment:
For these node names, I'm thinking we can maybe do something smarter to avoid copying the node name twice.

```suggestion
                ci = sdscatfmt(ci, " [%i->-", j);
                ci = sdscatlen(ci, server.cluster->migrating_slots_to[j]->name, CLUSTER_NAMELEN);
                ci = sdscat(ci, "]");
```